tests/suites/test_suite_psa_crypto_slot_management.function

```diff
@@ -643,12 +643,21 @@ void invalid_handle( )
     TEST_ASSERT( handle1 != 0 );
 
     /* Attempt to close and destroy some invalid handles. */
-    TEST_EQUAL( psa_close_key( 0 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_close_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_close_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( 0 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
+    if( handle1 - 1 != 0 )
+    {
+        TEST_EQUAL( psa_close_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
+        TEST_EQUAL( psa_destroy_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
+    }
+    if( handle1 + 1 != 0 )
+    {
+        TEST_EQUAL( psa_close_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
+        TEST_EQUAL( psa_destroy_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
+    }
+
+    /* 0 is special: it isn't a valid handle, but close/destroy
+     * succeeds on it. */
+    TEST_EQUAL( psa_close_key( 0 ), PSA_SUCCESS );
+    TEST_EQUAL( psa_destroy_key( 0 ), PSA_SUCCESS );
```

Comment:
Do we need these checks here? They don't use handle1 at all. We get equivalent coverage from the close_invalid() and destroy_invalid() tests (because they are called with handle 0), don't we?

```
psa_destroy_key(0)
destroy_invalid:0:PSA_SUCCESS

psa_close_key(0)
close_invalid:0:PSA_SUCCESS
```

Reply:
Yes, there's some redundancy between this test case and the `xxx_invalid` ones in `psa_crypto.function`. I don't think this redundancy is harmful so I didn't bother to remove it.

Reply:
Redundancy was added by this PR. Redundancy is harmful to maintainability, as there are more places to update if behavior changes.

Reply:
No, the redundancy was there before, and I kept it to avoid scope creep. But ok, I'll fix it.

Reply:
I've consolidated the two test functions.